purl_sync/review/views.py

```diff
@@ -0,0 +1,616 @@
+        """
+        You can GET from your inbox to read your latest messages
+        (client-to-server; this is like reading your social network stream)
+        """
+        actor = None
+        notes = Note.objects.filter(user=actor)
+        return JsonResponse(
+            {
+                "notes": ap_collection(notes),
+            },
+            content_type="application/activity+json",
+        )
+
+    def post(self, request, *args, **kwargs):
+        """
+        You can POST to someone's inbox to send them a message
+        (server-to-server / federation only... this is federation!)
+        """
+        activity = create_activity_obj(request.body)
+        if activity:
+            activity.handler()
+        else:
+            return JsonResponse(
+                {},
+                content_type=AP_CONTENT_TYPE,
+            )
+
+
+@method_decorator(is_valid_header, name="dispatch")
+class PurlOutbox(View):
+    def get(self, request, *args, **kwargs):
+        """GET from someone's outbox to see what messages they've posted
+        (or at least the ones you're authorized to see).
+        (client-to-server and/or server-to-server)"""
+
+        actor = None
+        notes = Note.objects.filter(author=actor)
+        return JsonResponse(
+            [note.to_ap() for note in notes],
+            content_type=AP_CONTENT_TYPE,
+        )
+
+    def post(self, request, *args, **kwargs):
+        """You can POST to your outbox to send messages to the world (client-to-server)"""
+
+
+def redirect_repository(request, repository_id):
+    try:
+        repo = Repository.objects.get(id=repository_id)
+    except Repository.DoesNotExist:
+        raise Http404("Repository does not exist")
+    return redirect(repo.url)
+
+
+class UserFollowing(View):
+    def get(self, request, *args, **kwargs):
+        followings = Follow.objects.filter(person__user__username=self.kwargs["username"])
+        return JsonResponse(
+            [full_reverse(following) for following in followings],
+            content_type="application/activity+json",
+        )
+
+
+class PurlFollowers(View):
+    def get(self, request, *args, **kwargs):
+        followers = Follow.objects.filter(purl__string=self.kwargs["purl_string"])
+        return JsonResponse(
+            [full_reverse(following) for following in followers],
+            content_type=AP_CONTENT_TYPE,
+        )
+
+
+@csrf_exempt
+def token(request):
+    payload = json.loads(request.body)
+    r = requests.post(
+        "http://127.0.0.1:8000/o/token/",
+        headers={"content-type": "application/x-www-form-urlencoded"},
+        data={
+            "grant_type": "password",
+            "username": payload["username"],
+            "password": payload["password"],
+            "client_id": CLIENT_ID,
+            "client_secret": CLIENT_SECRET,
+        },
+    )
+    return JsonResponse(json.loads(r.content), status=r.status_code, content_type=AP_CONTENT_TYPE)
```

Comment:
@TG1999 @pombredanne Is there any better way to pass client_id and client_secret instead of duplicate the requests again .